### PR TITLE
circle.yml: Use ./setup.py docs

### PR DESCRIPTION
**Pull request by nemaniarjun**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

Removes duplicate code

closes https://github.com/coala/coala/issues/3075

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -23,5 +23,4 @@ test:
     - bash .misc/deploy.coverage.sh:
         parallel: true
     - make -C docs clean
-    - sphinx-apidoc -f -o docs --no-toc coalib
-    - make -C docs html SPHINXOPTS=-W
+    - python setup.py docs
```
Comment (Mixih) [inline circle.yml]: Not a full duplication. We probably want -W for ci builds so warnings become errors. @sils your opinion is requested

Comment (jayvdb) [inline circle.yml]: Yes, `setup.py` doesnt currently set `SPHINXOPTS=-W`. The first option is for `setup.py` to always set `SPHINXOPTS=-W`.  If that causes problems, we can find better solution.

Comment (nemaniarjun) [inline circle.yml]: Done :)

**Comment by Mixih** [comment]
ack 7c9349

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**